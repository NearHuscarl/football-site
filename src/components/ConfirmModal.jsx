import React from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-modal';

const ConfirmModal = (props) => {
	const { isOpen, onRequestClose, prompt, onConfirm, onCancel } = props;
	return (
		<Modal
			isOpen={isOpen}
			onRequestClose={onRequestClose}
			contentLabel="Confirm Delete"
			closeTimeoutMS={200}
			className="modal"
			ariaHideApp={false}>
			<h3 className="modal__title">{prompt}</h3>
			<button
				type="button"
				className="button modal__button"
				onClick={onConfirm}>
				yes
			</button>
			<button
				type="button"
				className="button modal__button"
				onClick={onCancel}>
				no
			</button>
		</Modal>
	);
};

ConfirmModal.propTypes = {
	isOpen: PropTypes.bool.isRequired,
	onRequestClose: PropTypes.func,
	prompt: PropTypes.string.isRequired,
	onConfirm: PropTypes.func.isRequired,
	onCancel: PropTypes.func.isRequired,
};

ConfirmModal.defaultProps = {
	onRequestClose: () => {},
};

export default ConfirmModal;
