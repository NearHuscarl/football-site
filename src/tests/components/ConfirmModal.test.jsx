import React from 'react';
import { shallow } from 'enzyme';
import ConfirmModal from '../../components/ConfirmModal';

test('should render ConfirmModal correctly', () => {
	const wrapper = shallow(
		<ConfirmModal
			isOpen={false}
			prompt="are you sure?"
			onConfirm={jest.fn()}
			onCancel={jest.fn()}
		/>,
	);
	expect(wrapper).toMatchSnapshot();
});
