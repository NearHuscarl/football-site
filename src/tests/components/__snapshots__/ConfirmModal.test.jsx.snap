// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ConfirmModal correctly 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Confirm Delete"
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h3
    className="modal__title"
  >
    are you sure?
  </h3>
  <button
    className="button modal__button"
    onClick={[MockFunction]}
    type="button"
  >
    yes
  </button>
  <button
    className="button modal__button"
    onClick={[MockFunction]}
    type="button"
  >
    no
  </button>
</Modal>
`;
